paddle/fluid/framework/channel_impl.h

```diff
@@ -167,16 +170,20 @@ bool ChannelImpl<T>::Send(T *item) {
     buf_.push_back(std::move(*item));
     // Release lock and return true
     lock.unlock();
-    return send_return(true);
+    send_return();
+    return;
   }
 
   // Block on channel, because some receiver will complete
   // the operation for us
   auto m = std::make_shared<QueueMessage>(item);
   sendq.push_back(m);
   m->Wait(lock);
-  // TODO(abhinavarora) Should panic on closed channel
-  return send_return(!m->chan_closed);
+  if (m->chan_closed) {
+    send_return();
+    PADDLE_THROW("Cannot send on closed channel");
+  }
+  send_return();
```

Comment:
Do you need to unlock here?

Reply:
Good catch, the lock needs to be unlocked here. Thank you for pointing this out.